yarn-project/end-to-end/src/cheat_codes.ts

```diff
@@ -0,0 +1,169 @@
+    return parseInt(res.result, 16);
+  }
+
+  /**
+   * Get the current timestamp
+   * @returns The current timestamp
+   */
+  public async timestamp(): Promise<number> {
+    const res = await this.rpcCall('eth_getBlockByNumber', `["latest", true]`);
+    return parseInt(res.result.timestamp, 16);
+  }
+
+  /**
+   * Get the current chainId
+   * @param numberOfBlocks - The number of blocks to mine
+   * @returns The current chainId
+   */
+  public async mine(numberOfBlocks = 1): Promise<void> {
+    const res = await this.rpcCall('anvil_mine', `[${numberOfBlocks}]`);
+    if (res.error) throw new Error(`Error mining: ${res.error.message}`);
+    this.logger(`Mined ${numberOfBlocks} blocks`);
+  }
+
+  /**
+   * Set the next block timestamp
+   * @param timestamp - The timestamp to set the next block to
+   */
+  public async setNextBlockTimestamp(timestamp: number): Promise<void> {
+    const res = await this.rpcCall('anvil_setNextBlockTimestamp', `[${timestamp}]`);
+    if (res.error) throw new Error(`Error setting next block timestamp: ${res.error.message}`);
+    this.logger(`Set next block timestamp to ${timestamp}`);
+  }
+
+  // Good basis for the remaining functions:
+  // https://github.com/foundry-rs/foundry/blob/master/anvil/core/src/eth/mod.rs
+}
+
+/**
+ * A class that provides utility functions for interacting with the L2 chain.
+ */
+class L2CheatCodes {
+  constructor(
+    /**
+     * The RPC client to use for interacting with the chain
+     */
+    public aztecRpc: AztecRPC,
+    /**
+     * The circuits wasm module used for pedersen hashing
+     */
+    public wasm: CircuitsWasm,
+    /**
+     * The L1 cheat codes.
+     */
+    public l1: L1CheatCodes,
+    /**
+     * The logger to use for the l2 cheatcodes
+     */
+    public logger = createDebugLogger('aztec:cheat_codes:l2'),
+  ) {}
+
+  /**
+   * Computes the slot value for a given map and key.
+   * @param baseSlot - The base slot of the map (specified in noir contract)
+   * @param key - The key to lookup in the map
+   * @returns The storage slot of the value in the map
+   */
+  public computeSlotInMap(baseSlot: Fr | bigint, key: Fr | bigint): Fr {
+    // Based on `at` function in
+    // aztec3-packages/yarn-project/noir-contracts/src/contracts/noir-aztec/src/state_vars/map.nr
+    return Fr.fromBuffer(
+      pedersenPlookupCommitInputs(
+        this.wasm,
+        [toFr(baseSlot), toFr(key)].map(f => f.toBuffer()),
+      ),
+    );
+  }
+
+  /**
+   * Get the current blocknumber
+   * @returns The current block number
+   */
+  public async blockNumber(): Promise<number> {
+    return await this.aztecRpc.getBlockNum();
+  }
+
+  /**
+   * Loads the value stored at the given slot in the public storage of the given contract.
+   * @param who - The address of the contract
+   * @param slot - The storage slot to lookup
+   * @returns The value stored at the given slot
+   */
+  public async loadPublic(who: AztecAddress, slot: Fr | bigint): Promise<Fr> {
```

Comment:
load public is a bit ambiguous as a name, load public slot it more explicit
who could also just be address.

Reply:
Were using this naming to make it closer to foundry cheatcodes.